packages/core/src/interpreter.ts

```diff
@@ -818,12 +816,11 @@ export class Interpreter<
   }
 
   private stopChild(childId: string): void {
-    const child = this.children.get(childId);
+    const child = this.state.children[childId];
     if (!child) {
       return;
     }
 
-    this.children.delete(childId);
     this.forwardTo.delete(childId);
     delete this.state.children[childId];
```

Comment:
This probably should be refactored to avoid mutation of `this.state` - I think most people treat this as immutable. Maybe grabbing the actor ref to stop within machine APIs~ and embedding that in the action would help? This way we wouldn't have to access this here by ID

Reply:
Not a good idea; prevents them from getting GC'd. This is one of those cases where we might need to allow these to be mutated.

Reply:
Let's address this in a separate PR. This isn't part of this PR's changes.

Reply:
Do we have any ticket anywhere to track this?

Reply:
#2851